chainermn/communicators/_base.py

```diff
@@ -15,6 +19,19 @@ def rank(self):
     def size(self):
         return self.mpi_comm.size
 
+    def send(self, array, dest, tag):
+        shape = numpy.array(array.shape, dtype='int')
+        buf = _memory_utility.array_to_buffer_object(array.astype('float32'))
+        self.mpi_comm.Send([shape, mpi4py.MPI.INT], dest=dest, tag=tag)
+        self.mpi_comm.Send(buf, dest=dest, tag=tag)
+
+    def recv(self, source, tag):
+        shape = numpy.empty(2, dtype='int')
+        self.mpi_comm.Recv([shape, mpi4py.MPI.INT], source=source, tag=tag)
+        buf = numpy.empty(shape[0] * shape[1], dtype='float32')
```

Comment:
Seemingly this code only supports two dimensional arrays. We need to handle arrays of different dimensions. For example, when using CNNs, activation is four dimensional arrays.